resources/ts/Pages/Integrations/New.tsx

```diff
@@ -323,6 +336,39 @@ const New = ({ subscriptions }: Props) => {
               }
               error={errors.agreement}
             />
+            {isCouponFieldVisible && (
+              <>
+                <FormElement
+                  label={t("integration_form.coupon")}
+                  labelPosition="right"
+                  labelSize="base"
+                  labelWeight="normal"
+                  component={
+                    <input
+                      type="checkbox"
+                      name="hasCoupon"
+                      className="text-publiq-blue-dark focus:ring-publiq-blue-dark rounded-sm"
+                      checked={hasCoupon}
+                      onChange={() => setHasCoupon(!hasCoupon)}
```

Comment:
```suggestion
                      onChange={() => setHasCoupon(prev => !prev)}
```
This is safer to use

Reply:
https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state